LLama/Native/LLamaModelParams.cs

```diff
@@ -38,7 +38,7 @@ public unsafe struct LLamaModelParams
         // as NET Framework 4.8 does not play nice with the LlamaProgressCallback type
         public IntPtr progress_callback;
 #else
-        public LlamaProgressCallback progress_callback;
+        public LlamaProgressCallback? progress_callback;

```

Comment:
Recently I'm wondering if it's better to use a modern progress interface here ([`IProgress<T>`](https://learn.microsoft.com/en-us/dotnet/api/system.iprogress-1?view=net-8.0)). In this way the `LLamaContextHandle` object would not be exposed to users who works with LLamaSharp high-level APIs. For us (or other experienced users), we won't use this parameter in a wrong way. But for some users who know little about our backend, it might be confusing.

I prefer to switching to `IProgress<float>` smoothly, maybe not in this PR though. :)

Reply:
`IProgress` is a good idea 👍 

I intentionally didn't expose the raw progress callback to the user in this PR because it does feel a bit ugly - it's combining progress and cancellation together. We should probably expose it just in case someone wants to use it, but have better options available for most users.

I'll have a look at adding an `IProgress` parameter to `LoadFromFileAsync` now.